src/main/java/org/pesmypetcare/webservice/builders/Path.java

```diff
@@ -0,0 +1,175 @@
+package org.pesmypetcare.webservice.builders;
+
+import org.springframework.lang.NonNull;
+
+/**
+ * @author Santiago Del Rey
+ */
+public class Path {
+    private static final DocumentPath DOCUMENT_PATH = new DocumentPath();
+    private static final CollectionPath COLLECTION_PATH = new CollectionPath();
+
+    private Path() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Builds the path to the desired document or collection.
+     * <p>
+     * There must be as many IDs as documents should be crossed until the desired file is reached. Also, the IDs must
+     * be in the same order they need to be accessed to reach the final document or collection.
+     * <p>
+     * Here are some examples of how the method should be used:
+     * <blockquote><pre>
+     *     //Path to the Dogs group
+     *     String path = Path.ofDocument(Collections.group_names, "Dogs");
+     *
+     *     //Path to the Huskies forum in the Dogs group
+     *     String path = Path.ofDocument(Collections.forumsNames, "Dogs", "Huskies");
+     *
+     *     //Path to the pet Rex of the user with ID yIoqQ6OepwRaadAOz2EdeIhikOX2
+     *     String path = Path.ofDocument(Collections.pets, "yIoqQ6OepwRaadAOz2EdeIhikOX2", "Rex");
+     * </pre></blockquote>
+     *
+     * @param collection The collection where the document is stored
+     * @param ids The ids the builder should use, going from the most outer one to the inner one
+     * @return The path to a document
+     */
+    @NonNull
+    public static String ofDocument(@NonNull Collections collection, @NonNull String... ids) {
+        checkProvidedIds(collection, ids);
+        switch (collection) {
+            case groups:
+            case groups_names:
+            case tags:
+            case used_usernames:
+            case users:
+            case medals:
+                return DOCUMENT_PATH.buildOneLevelPath(collection, ids).toString();
+            case forums:
+            case forum_names:
+            case pets:
+            case userMedals:
+            case members:
+                return DOCUMENT_PATH.buildTwoLevelPath(collection, ids).toString();
+            case messages:
+            case kcals:
+            case trainings:
+            case kcalsAverages:
+            case weekTrainings:
+            case meals:
+            case weights:
+            case exercises:
+            case washes:
+            case vaccinations:
+            case illnesses:
+            case medications:
+            case vet_visits:
+                return DOCUMENT_PATH.buildThreeLevelPath(collection, ids).toString();
```

Comment:
delete the ones that don't exist anymore

Reply:
Done